### PR TITLE
Drops main compile dependency on io.zipkin.zipkin2:zipkin 

### PR DESCRIPTION
**Pull request by adriancole**
This uses Brave's remote endpoint api.

See openzipkin/brave#747

Depends on #1328

### CODE REVIEW

File: zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java [hunk 1]
```diff
@@ -50,9 +50,20 @@
      * Use this singleton when building a {@link brave.Tracing} instance for use with
      * {@link HttpTracingService} or {@link HttpTracingClient}.
      *
+     * <p>If you need to customize the context, use {@linkplain #newBuilder()} instead.
```
Comment (trustin) [inline zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java]: `@linkplain` -> `@link`

File: zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java [hunk 2]
```diff
@@ -50,9 +50,20 @@
...
-    public static final CurrentTraceContext INSTANCE = new RequestContextCurrentTraceContext();
+    public static final CurrentTraceContext DEFAULT = new RequestContextCurrentTraceContext(new Builder());
+
+    /**
+     * Singleton retained for backwards compatibility, but replaced by {@link #DEFAULT}.
+     *
+     * @deprecated Please use {@link #DEFAULT} or {@link #newBuilder()} to customize an instance.
+     */
+    @Deprecated
+    public static final CurrentTraceContext INSTANCE = new RequestContextCurrentTraceContext(new Builder());
```
Comment (trustin) [inline zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java]: Could be just `= DEFAULT`?

File: zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java (hunk 1 above)
Comment (adriancole) [reply zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java]: addressed in the other PR (will rebase this once merged)

File: zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java (hunk 2 above)
Comment (adriancole) [reply zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java]: addressed in the other PR (will rebase this once merged)

**Comment by trustin** [comment]
@adriancole Please rebase. :-)

**Comment by adriancole** [comment]
done!

File: zipkin/src/main/java/com/linecorp/armeria/client/tracing/HttpTracingClient.java [hunk 3]
```diff
@@ -120,10 +119,14 @@ private void setRemoteEndpoint(Span span, RequestLog log) {
 
         final SocketAddress remoteAddress = log.context().remoteAddress();
         final InetAddress address;
+        final int port;
         if (remoteAddress instanceof InetSocketAddress) {
-            address = ((InetSocketAddress) remoteAddress).getAddress();
+            InetSocketAddress socketAddress = (InetSocketAddress) remoteAddress;
```
Comment (trustin) [inline zipkin/src/main/java/com/linecorp/armeria/client/tracing/HttpTracingClient.java]: nit: missing `final`

**Comment by trustin** [comment]
Thanks, @adriancole !

**Pull request merged**